runtime/pkg/jsonval/jsonval.go

```diff
@@ -0,0 +1,382 @@
+package jsonval
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"math/big"
+	"net"
+	"reflect"
+	"time"
+	"unicode/utf8"
+
+	"github.com/google/uuid"
+	"github.com/marcboeker/go-duckdb"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+)
+
+// ToString is similar to ToValue, but finally converts the value to a JSON string.
+func ToString(v any, t *runtimev1.Type) (string, error) {
+	v, err := ToValue(v, t)
+	if err != nil {
+		return "", err
+	}
+	if v == nil {
```

Comment:
Is it fine to return an empty string when v is nil in all cases ?

Reply:
"All cases" are a little hard to think through here, but it was needed for the current case – serializing the value to a CSV record. It would be incorrect to output `nil` since that could be misinterpreted as a real string value.

I think you're right this doesn't seem generic. Since it's only currently used for CSV serialization, I removed this function and inlined it in the CSV serializer instead.